test/performance/odp_packet_gen.c

```diff
@@ -741,6 +775,34 @@ static int start_pktios(test_global_t *global)
 		global->pktio[i].started = 1;
 	}
 
+	if (!test_options->wait_sec)
+		return 0;
+
+	/* Wait until all links are up */
+	for (i = 0; i < num_pktio; i++) {
+		uint32_t link_wait = 0;
```

Comment:
Now each pktio may wait separately up to the max wait time instead of there being max total wait time as one would expect. I would propose moving this outside of the loop as all the pktios have already been started when entering the loop.